File: frontend/src/domain/transcription/value-objects/transcript-text.ts
/**
 * Transcript text value object.
 */

export class TranscriptText {
    private constructor(public readonly value: string) {
        if (!value.trim()) {
            throw new Error('Transcript text cannot be empty');
        }
    }

    static create(text: string): TranscriptText {
        const normalized = text.split(/\s+/).join(' ').trim();
        return new TranscriptText(normalized);
    }

    get wordCount(): number {
        return this.value.split(/\s+/).length;
    }

    get characterCount(): number {
        return this.value.length;
    }

    equals(other: TranscriptText): boolean {
        return this.value === other.value;
    }
}

File: backend/streamcraft/domain/transcription/value_objects/transcript_text.py
"""Transcript text value object."""

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class TranscriptText:
    """Validated transcript text."""

    value: str

    def __post_init__(self) -> None:
        """Validate transcript text."""
        if not self.value.strip():
            raise ValueError("Transcript text cannot be empty")

    @classmethod
    def create(cls, text: str) -> "TranscriptText":
        """Create with normalization."""
        normalized = " ".join(text.split())  # Normalize whitespace
        return cls(value=normalized)

    @property
    def word_count(self) -> int:
        """Get word count."""
        return len(self.value.split())

    @property
    def character_count(self) -> int:
        """Get character count."""
        return len(self.value)
